File: snippets/fsharp/System/TupleT1,T2,T3,T4,T5/System.Collections.IStructuralComparable.CompareTo/compareto1.fs
module compareto1

// <Snippet1>
open System

// Organization of runningBacks 5-tuple:
//    Component 1: Player name
//    Component 2: Number of games played
//    Component 3: Number of attempts (carries)
//    Component 4: Number of yards gained 
//    Component 5: Number of touchdowns   
let runningBacks =
    [| Tuple.Create("Payton, Walter", 190, 3838, 16726, 110)
       Tuple.Create("Sanders, Barry", 153, 3062, 15269, 99)
       Tuple.Create("Brown, Jim", 118, 2359, 12312, 106)
       Tuple.Create("Dickerson, Eric", 144, 2996, 13259, 90)
       Tuple.Create("Faulk, Marshall", 176, 2836, 12279, 100) |]

// Display the array in unsorted order.
printfn "The values in unsorted order:"
for runningBack in runningBacks do
    printfn $"{runningBack}"
printfn ""

// Sort the array
Array.Sort runningBacks

// Display the array in sorted order.
printfn "The values in sorted order:"
for runningBack in runningBacks do
    printfn $"{runningBack}"
// The example displays the following output:
//       The values in unsorted order:
//       (Payton, Walter, 190, 3838, 16726, 110)
//       (Sanders, Barry, 153, 3062, 15269, 99)
//       (Brown, Jim, 118, 2359, 12312, 106)
//       (Dickerson, Eric, 144, 2996, 13259, 90)
//       (Faulk, Marshall, 176, 2836, 12279, 100)
//       
//       The values in sorted order:
//       (Brown, Jim, 118, 2359, 12312, 106)
//       (Dickerson, Eric, 144, 2996, 13259, 90)
//       (Faulk, Marshall, 176, 2836, 12279, 100)
//       (Payton, Walter, 190, 3838, 16726, 110)
//       (Sanders, Barry, 153, 3062, 15269, 99)
// </Snippet1>

File: snippets/csharp/System/TupleT1,T2,T3,T4,T5,T6,T7,TRest/System.Collections.IStructuralComparable.CompareTo/compareto1.cs
// <Snippet1>
using System;

public class Example
{
    public static void Main()
    {
        // Create array of 8-tuple objects containing prime numbers.
        Tuple<int, int, int, int, int, int, int, Tuple<int>>[] primes =
                       { new Tuple<int, int, int, int, int, int, int,
                           Tuple<int>>(2, 3, 5, 7, 11, 13, 17, new Tuple<int>(19)),
                       new Tuple<int, int, int, int, int, int, int,
                           Tuple<int>>(23, 29, 31, 37, 41, 43, 47, new Tuple<int>(55)),
                       new Tuple<int, int, int, int, int, int, int,
                           Tuple<int>>(3, 2, 5, 7, 11, 13, 17, new Tuple<int>(19)) };
        // Display 8-tuples in unsorted order.
        foreach (var prime in primes)
            Console.WriteLine(prime.ToString());
        Console.WriteLine();

        // Sort the array and display its 8-tuples.
        Array.Sort(primes);
        foreach (var prime in primes)
            Console.WriteLine(prime.ToString());
    }
}
// The example displays the following output:
//       (2, 3, 5, 7, 11, 13, 17, 19)
//       (23, 29, 31, 37, 41, 43, 47, 55)
//       (3, 2, 5, 7, 11, 13, 17, 19)
//
//       (2, 3, 5, 7, 11, 13, 17, 19)
//       (3, 2, 5, 7, 11, 13, 17, 19)
//       (23, 29, 31, 37, 41, 43, 47, 55)
// </Snippet1>
